qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,964 @@
+                      "from the set origin",
+        )
+
+        self.add_parameter(
+            "wait",
+            get_cmd=None,
+            set_cmd="WT {}",
+            unit="ms",
+            vals=Multiples(min_value=2, max_value=2147483646, divisor=2),
+            docstring="controller will wait for the amount of "
+            "time specified before executing the next "
+            "command",
+        )
+
+        self._set_default_update_time()
+        self.add_submodule("motor_a", Motor(self, "A"))
+        self.add_submodule("motor_b", Motor(self, "B"))
+        self.add_submodule("motor_c", Motor(self, "C"))
+        self.add_submodule("plane_ab", VectorMode(self, "AB"))
+        self.add_submodule("plane_bc", VectorMode(self, "BC"))
+        self.add_submodule("plane_ac", VectorMode(self, "AC"))
+
+        self.connect_message()
+
+    def _set_default_update_time(self) -> None:
+        """
+        sets sampling period to default value of 1000. sampling period affects
+        the AC, AS, AT, DC, FA, FV, HV, JG, KP, NB, NF, NZ, PL, SD, SP, VA,
+        VD, VS, WT commands.
+        """
+        self.write("TM 1000")
+
+    def _get_absolute_position(self) -> Dict[str, int]:
+        """
+        gets absolution position of the motors from the defined origin
+        """
+        result = dict()
+        data = self.ask("PA ?,?,?").split(" ")
+        result["A"] = int(data[0][:-1])
+        result["B"] = int(data[1][:-1])
+        result["C"] = int(data[2])
+
+        return result
+
+    def end_program(self) -> None:
+        """
+        ends the program
+        """
+        self.write("EN")
+
+    def define_position_as_origin(self) -> None:
+        """
+        defines current motors position as origin
+        """
+        self.write("DP 0,0,0")
+
+    def tell_error(self) -> str:
+        """
+        reads error
+        """
+        return self.ask("TC1")
+
+    def stop(self) -> None:
+        """
+        stop the motion of all motors
+        """
+        self.write("ST")
+
+    def abort(self) -> None:
+        """
+        aborts motion and the program operation
+        """
+        self.write("AB")
+
+    def motors_off(self) -> None:
+        """
+        turn all motors off
+        """
+        self.write("MO")
+
+    def begin_motors(self) -> None:
+        """
+        begin motion of all motors simultaneously
+        """
+        self.write("BG")
+
+    def wait_till_motion_complete(self) -> None:
+        """
+        this method waits for the motion on all motors to complete
+        """
+        try:
+            while (
+                int(float(self.ask("MG _BGA")))
+                or int(float(self.ask("MG _BGB")))
+                or int(float(self.ask("MG _BGC")))
+            ):
+                pass
+        except KeyboardInterrupt:
+            self.abort()
+            self.motors_off()
+
+
+class Arm:
+    """
+    Module to control probe arm
```

Comment:
this docstring would be much more helpful if it wouls include information on what kind of arm/chip is assumed here.